include/psa/crypto_extra.h

```diff
@@ -1901,6 +1901,342 @@ static inline struct psa_pake_operation_s psa_pake_operation_init( void )
+ *
+ * \note                        It is strictly required that all input
+ *                              parameters and the restart context be the same
+ *                              on successive calls for the same operation,
+ *                              but output parameters need not be the same;
+ *                              they must not be used until the function
+ *                              finally returns 0.
+ *
+ *                              This only applies to functions whose
+ *                              documentation mentions they may return
+ *                              #PSA_ERROR_IN_PROGRESS.
+ *
+ * \param max_ops               The maximum number of basic operations to
+ *                              be done in a row. Passing zero here has the
+ *                              same effect as passing the maximum value - i.e
+ *                              the operation will be done in one go. Lower
+ *                              (non-zero) values mean signature functions
+ *                              will block for a lesser maximum amount of time
+ *                              and conversely larger values will mean
+ *                              blocking for a larger maximum amount of time.
+ *                              The effect of changing this value will depend
+ *                              on the key used, the curve used and the
+ *                              underlying implementation, so we provide /c
+ *                              psa_sign_interruptible_get_num_ops() in order
+ *                              to help with tuning this value.
+ *
+ */
+void psa_sign_interruptible_set_max_ops( size_t max_ops );
+
+/**
+ * \brief                       Get the number of basic operations that an
+ *                              operation took to complete.
+ *
+ *                              This is a helper provided to help you tune the
+ *                              value passed to \c
+ *                              psa_sign_interruptible_set_max_ops().
+ *
+ *                              This only applies to functions whose
+ *                              documentation mentions they may return
+ *                              #PSA_ERROR_IN_PROGRESS.
+ *
+ * \param ctx                   The \c psa_sign_interruptible_context to use.
+ *                              This must be initialized first.
+ *
+ */
+size_t psa_sign_interruptible_get_num_ops(
+                                          psa_sign_interruptible_context *ctx );
+
+/**
+ * \brief                       Start signing a hash or short message with a
+ *                              private key, in an interruptible manner.
+ *
+ * \see                         \c psa_sign_hash_complete()
+ *
+ * \note                        This function combined with \c
+ *                              psa_sign_hash_complete() is equivalent to
+ *                              \c psa_sign_hash() but it can return early and
```

Comment:
Ambiguous syntax: does "it" refer to "this function" or `psa_sign_hash_complete()`? It's easy to conclude that it's the later by noticing this function can't return `IN_PROGRESS`, but we should still avoid requiring the reader to do this mental gymnastic.